### DISCUSSION TITLE
Manual/Tags

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#87] [Notes]
# Manual/Tags

https://docs.unity3d.com/Manual/Tags.html

**Comment by vertxxyz** [comment]
The **EditorOnly** tag only works with instances in scene, ignoring tagged assets is unsupported.

**Comment by mvi** [comment]
The built in **EditorOnly** tag has special behaviour. At build time any objects tagged EditorOnly will be stripped out (including child objects regardless of whether the child is also tagged EditorOnly). This behaviour only occurs at build time, EditorOnly tagged objects will remain when pressing Play in the Editor.  
Marking objects as EditorOnly is useful for development helpers (such as debug text) that you may want in Editor but not in builds.  
If you do want to strip EditorOnly objects in editor at Play time as well as at build time there are a couple of simple approaches:  
- Write a script which destroys its gameObject on `Awake`.  
- Write a [`PostProcessSceneAttribute`](https://docs.unity3d.com/ScriptReference/Callbacks.PostProcessSceneAttribute.html) that strips all EditorOnly objects. Note that PostProcessSceneAttributes are called on Play as well as at build time.
